core/templates/pages/exploration-editor-page/editor-navigation/editor-navbar-breadcrumb.component.spec.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for editorNavbarBreadcrumb.
+ */
+
+describe('Editor Navbar Breadcrumb directive', function() {
+  var $rootScope = null;
+  var $scope = null;
+  var ExplorationTitleService = null;
+  var FocusManagerService = null;
+  var RouterService = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    $rootScope = $injector.get('$rootScope');
+    ExplorationTitleService = $injector.get('ExplorationTitleService');
+    FocusManagerService = $injector.get('FocusManagerService');
+    RouterService = $injector.get('RouterService');
+
+    ExplorationTitleService.init('Exploration Title Example Very Long');
+
+    $scope = $rootScope.$new();
+    var ctrl = $componentController('editorNavbarBreadcrumb', {
+      $scope: $scope
+    });
+    ctrl.$onInit();
+  }));
+
+  it('should evaluate scope properties after controller initialization',
+    function() {
+      expect($scope.navbarTitle).toBe(null);
+    });
+
+  it('should go to settings tabs and focus on exploration title input' +
+    ' when editing title', function() {
+    spyOn(RouterService, 'navigateToSettingsTab');
+    spyOn(FocusManagerService, 'setFocus');
+
+    $scope.editTitle();
+
+    expect(RouterService.navigateToSettingsTab).toHaveBeenCalled();
+    expect(FocusManagerService.setFocus).toHaveBeenCalledWith(
+      'explorationTitleInputFocusLabel');
+  });
+
+  it('should get an empty current tab name when there is no active tab',
+    function() {
+      spyOn(RouterService, 'getActiveTabName').and.returnValue(null);
+      expect($scope.getCurrentTabName()).toBe('');
+    });
+
+  it('should get current tab name when there is an active tab', function() {
+    spyOn(RouterService, 'getActiveTabName').and.returnValue('settings');
+    expect($scope.getCurrentTabName()).toBe('Settings');
```

Comment:
The case of the 2 strings are different. Is this expected?

Reply:
Yes!